cleanup/Jenkinsfile

```diff
@@ -0,0 +1,63 @@
+import groovy.json.JsonSlurper
+
+pipeline {
+    agent none
+    options {
+        timeout(time: 1, unit: 'HOURS')
+        disableConcurrentBuilds()
+    }
+    stages {
+        stage('Cleanup') {
+            parallel {
+                stage('Cleanup Windows temp directory') {
+                    agent {
+                        label 'windows'
+                    }
+                    steps {
+                        script {
+                            cleanDirs('tmp/yarn', '\"%temp%\"', '(yarn--*)')
+                            cleanDirs('tmp/plugin-download', '\"%temp%\"', '(theia-plugin-download*)')
+                            cleanDirs('tmp/lighthouse', '\"%temp%\"', '(lighthouse.*)')
+                            cleanDir('appdata/local/electron', '\"%LocalAppData%\"\\electron\\Cache')
+                            cleanYarnCache('appdata/local/yarn', '\"%LocalAppData%\"\\Yarn\\Cache')
```

Comment:
Currently I would suggest to run it periodically, so either daily or weekly. This will avoid that we keep on using more and more disk space and we don't have to rebuild caches all the time. 

Otherwise it is possible to launch a Job from a different Job: https://www.jenkins.io/doc/pipeline/steps/pipeline-build-step/
So we may trigger this to the regular Build and Next Builds as a last step as well.

Reply:
I think the configuration, to define when to run this Jenkins file, done in Jenkins itself?

Reply:
I would suggest doing one run ASAP after merge and then setting-up the job to run e.g. weekly, at a day/time unlikely to conflict with another CI run.